openapi-core/src/main/java/io/ballerina/openapi/core/GeneratorUtils.java

```diff
@@ -958,15 +963,22 @@ private static List<String> getUnusedTypeDefinitionNameList(Map<String, String>
         List<String> unusedTypeDefinitionNameList = new ArrayList<>();
         Path tmpDir = Files.createTempDirectory(".openapi-tmp" + System.nanoTime());
         writeFilesTemp(srcFiles, tmpDir);
-        if (Files.exists(tmpDir.resolve(CLIENT_FILE_NAME)) && Files.exists(tmpDir.resolve(TYPE_FILE_NAME)) &&
+        if ((Files.exists(tmpDir.resolve(CLIENT_FILE_NAME)) || Files.exists(tmpDir.resolve(SERVICE_FILE_NAME)))
+                && Files.exists(tmpDir.resolve(TYPE_FILE_NAME)) &&
                 Files.exists(tmpDir.resolve(BALLERINA_TOML))) {
-            SemanticModel semanticModel = getSemanticModel(tmpDir.resolve(CLIENT_FILE_NAME));
+            SemanticModel semanticModel = Files.exists(tmpDir.resolve(CLIENT_FILE_NAME)) ?
+                    getSemanticModel(tmpDir.resolve(CLIENT_FILE_NAME)) :
+                    getSemanticModel(tmpDir.resolve(SERVICE_FILE_NAME));
             List<Symbol> symbols = semanticModel.moduleSymbols();
             for (Symbol symbol : symbols) {
                 if (symbol.kind().equals(SymbolKind.TYPE_DEFINITION) || symbol.kind().equals(SymbolKind.ENUM)) {
                     List<Location> references = semanticModel.references(symbol);
                     if (references.size() == 1) {
                         unusedTypeDefinitionNameList.add(symbol.getName().get());
+                    } else if (references.size() == 2) {
```

Comment:
The purpose of setting `references.size() == 2` is for handling the unused cyclic reference in the types. If we check `references.size() >1` it will take other used references into cyclic checks.

Reply:
It makes sense. +1